OpenRA.Game/Network/GameSave.cs

```diff
@@ -0,0 +1,287 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2019 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using OpenRA.Primitives;
+using OpenRA.Server;
+
+namespace OpenRA.Network
+{
+	public class SlotClient
+	{
+		public Color Color;
+		public string Faction;
+		public int SpawnPoint;
+		public int Team;
+		public string Slot;
+		public string Bot;
+		public int BotControllerClientIndex;
+		public bool IsAdmin;
+
+		public string BotName;
+
+		public SlotClient() { }
+
+		public SlotClient(Session.Client client)
+		{
+			Color = client.Color;
+			Faction = client.Faction;
+			SpawnPoint = client.SpawnPoint;
+			Team = client.Team;
+			Slot = client.Slot;
+			Bot = client.Bot;
+			BotControllerClientIndex = client.BotControllerClientIndex;
```

Comment:
This is probably bogus - we need to remap the client index to a SlotClient index.
Not a problem for mission saves or even skirmish saves, but I expect this will cause problems in the far future of MP saves.

Reply:
I removed this dodgy serialization in favor of a documented workaround (see the `// HACK` comment in `DispatchOrders`) that only needs to be resolved if we ever implement MP saves.